spec/unit/puppet/provider/conf_setting/ruby_spec.rb

```diff
@@ -517,13 +517,41 @@ def validate_file(expected_content,tmpfile = tmpfile)
       expect(provider.value[0]).to eql(true)
     end
 
-    it "should be able to handle number value with number type specified" do
+    it "should be able to handle an integer value with number type specified" do
       resource = Puppet::Type::Hocon_setting.new(common_params.merge(
                                                      :setting => 'test_key_1.master', :value => 12, :type => 'number'))
       provider = described_class.new(resource)
       provider.create
       expect(provider.exists?).to be true
-      expect(provider.value[0]).to eql(12)
+      expect(provider.value[0].eql?(12)).to be(true)
+    end
+
+    it "should be able to handle a float value with number type specified" do
+      resource = Puppet::Type::Hocon_setting.new(common_params.merge(
+                                                     :setting => 'test_key_1.master', :value => 13.37, :type => 'number'))
+      provider = described_class.new(resource)
+      provider.create
+      expect(provider.exists?).to be true
+      expect(provider.value[0].eql?(13.37)).to be(true)
+    end
+
+    it "should be able to handle an Integer string value with number type specified" do
+      resource = Puppet::Type::Hocon_setting.new(common_params.merge(
+                                                     :setting => 'test_key_1.master', :value => '12', :type => 'number'))
+      provider = described_class.new(resource)
+      provider.create
+      expect(provider.exists?).to be true
+      #expect(provider.value[0]).to eql(12)
```

Comment:
Done and pushed

On Thu, Apr 30, 2015 at 1:20 PM, Preben Ingvaldsen <notifications@github.com
> wrote:

> In spec/unit/puppet/provider/conf_setting/ruby_spec.rb
> <https://github.com/puppetlabs/puppetlabs-hocon/pull/12#discussion_r29449776>
> :
>
> > +    it "should be able to handle a float value with number type specified" do
> > +      resource = Puppet::Type::Hocon_setting.new(common_params.merge(
> > +                                                     :setting => 'test_key_1.master', :value => 13.37, :type => 'number'))
> > +      provider = described_class.new(resource)
> > +      provider.create
> > +      expect(provider.exists?).to be true
> > +      expect(provider.value[0].eql?(13.37)).to be(true)
> > +    end
> > +
> > +    it "should be able to handle an Integer string value with number type specified" do
> > +      resource = Puppet::Type::Hocon_setting.new(common_params.merge(
> > +                                                     :setting => 'test_key_1.master', :value => '12', :type => 'number'))
> > +      provider = described_class.new(resource)
> > +      provider.create
> > +      expect(provider.exists?).to be true
> > +      #expect(provider.value[0]).to eql(12)
>
> I think we should remove this commented line.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppetlabs-hocon/pull/12/files#r29449776>.
>



-- 
Gary Larizza
Professional Services Engineer
Puppet Labs

http://garylarizza.com